lib/processes/proxy/proxy.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+var http = require('http');
+var url = require('url');
+
+var concat = require('concat-stream');
+var _ = require('lodash');
+var qs = require('qs');
+
+var modifyResponse = require('./cookie').modifyResponse;
+
+function buildRemoteRequestOptions(request, toPort) {
+  var uri = url.parse(request.url);
+  var opt = {
+    port: toPort,
+    path: uri.path,
+    method: request.method,
+    headers: request.headers
+  };
+
+  opt.headers.connection = 'keep-alive';
+  opt.headers['cache-control'] = 'no-store';
+  delete opt.headers['if-none-match'];
+  delete opt.headers['if-modified-since'];
+  return opt;
+}
+
+function trimHash(url) {
+  return url.split('#')[0];
+}
+
+function normalizeOptions(options) {
+  options.url = trimHash(options.url);
+  options.redirect = options.redirect === true || options.redirect === 'true';
+  return options;
+}
+
+var openRequests = [];
+var firstPage = true;
+var newPageOptions = {};
+
+function markNewPage(options, response, fromPort) {
+  var newPageOptions = normalizeOptions(options);
+  console.log('\n[System] Marking new page request with options: %j', newPageOptions);
+
+  openRequests.forEach(function(request) {
+    console.log('[System] Aborting request for: %s', request.path);
+    request.aborted = true;
+    request.abort();
+  });
+  openRequests = [];
+
+  if (options.redirect) {
```

Comment:
@EndangeredMassa I'm not sure if this will actually work. My current hope is that we can just blindly navigate to `testium.getNewPageUrl('/foobar')` and the redirect here will make the magic happen. Having it trigger via `GET` has the potential advantage that we don't need to implement a non-webdriver communication channel.